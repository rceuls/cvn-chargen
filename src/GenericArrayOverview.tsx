import { chunkArray } from './services/util';

const GenericArrayOverview = ({ data }: { data: string[] }) => {
  if (data) {
    const chonked = chunkArray(
      data.length > 2 ? data : ['', '', ...data],
      2
    ).reverse(); // most items first
    return (
      <>
        {chonked[0].map((eq, ix) => (
          <div className="row">
            <div className="col-sm">{eq}</div>
            {chonked[1].length !== ix ? (
              <div className="col-sm">{chonked[1][ix]}</div>
            ) : (
              <></>
            )}
          </div>
        ))}
        <hr />
      </>
    );
  } else {
    return <></>;
  }
};

export default GenericArrayOverview;
