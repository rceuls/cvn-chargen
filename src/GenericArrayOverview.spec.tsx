import renderer from 'react-test-renderer';
import GenericArrayOverview from './GenericArrayOverview';

it('renders correctly', () => {
  const tree = renderer
    .create(<GenericArrayOverview data={['a', 'b', 'c', 'd', 'e', 'f']} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
