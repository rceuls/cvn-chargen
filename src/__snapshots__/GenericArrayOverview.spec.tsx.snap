// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      d
    </div>
    <div
      className="col-sm"
    >
      a
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      e
    </div>
    <div
      className="col-sm"
    >
      b
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      f
    </div>
    <div
      className="col-sm"
    >
      c
    </div>
  </div>,
  <hr />,
]
`;
